mock/py/mockbuild/plugins/rpmautospec.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# vim:expandtab:autoindent:tabstop=4:shiftwidth=4:filetype=python:textwidth=0:
+# License: GPL2 or later see COPYING
+# Copyright (C) 2023 Stephen Gallagher <sgallagh@redhat.com>
+# Copyright (C) 2023 Nils Philippsen <nils@redhat.com>
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Optional, Union
+
+from rpmautospec_core import specfile_uses_rpmautospec
+
+from mockbuild.exception import ConfigError
+from mockbuild.trace_decorator import getLog, traceLog
+
+requires_api_version = "1.1"
+
+
+@traceLog()
+def init(plugins, conf, buildroot):
+    RpmautospecPlugin(plugins, conf, buildroot)
+
+
+class RpmautospecPlugin:
+    """Fill in release and changelog from git history using rpmautospec"""
+
+    @traceLog()
+    def __init__(self, plugins, conf, buildroot):
+        self.buildroot = buildroot
+        self.config = buildroot.config
+        self.opts = conf
+        self.log = getLog()
+
+        if "cmd_base" not in self.opts:
+            raise ConfigError("The 'rpmautospec_opts.cmd_base' is unset")
+
+        plugins.add_hook("pre_srpm_build", self.attempt_process_distgit)
+        self.log.info("rpmautospec: initialized")
+
+    @contextmanager
+    def as_root_user(self):
+        try:
+            self.buildroot.uid_manager.becomeUser(0, 0)
+            yield
+        finally:
+            self.buildroot.uid_manager.restorePrivs()
```

Comment:
Do we actually need this? As far as I know, we only actually needed to become root in order to install packages into the buildroot, but we switched below to using `buildroot.pkg_manager.install_as_root()` instead of managing the privilege escalation ourselves.

Reply:
Yeah, looks superfluous. I’ll remove it.